autonick.lua

```diff
@@ -0,0 +1,53 @@
+-- automatically nickname dwarves
+--[====[
+
+autonick
+========
+Gives dwarves unique nicknames chosen randomly from ``dfhack-config/autonick.txt``.
+
+One nickname per line.
+Empty lines, lines beginning with ``#`` and repeat entries are discarded.
```

Comment:
I just noticed that repeat entries are *not* discarded. They could be, though, if we add the names read from the config file to the `taken` set as we read them. Then any repeat entries will be properly ignored.

maybe rename `taken` to `seen` as well to reflect its new role.